integration/src/test/scala/com/spotify/scio/bigquery/TypedBigQueryIT.scala

```diff
@@ -190,10 +224,42 @@ class TypedBigQueryIT extends PipelineSpec with BeforeAndAfterAll {
         )
     }.waitUntilFinish()
 
+    runWithRealContext(options) { sc =>
+      sc.typedBigQuery[Record](avroTable) should containInAnyOrder(records)
+    }
+
+    // Due to Beam bug with automatic schema detection, can't parse nested record types as GenericRecords yet
+    // Todo remove assertThrows after fixing in Beam
+    assertThrows[UnresolvedUnionException] {
```

Comment:
basically Beam attempts to infer an Avro schema the selected BQ table here, and it doesn't work correctly if you have a nested record (capitalization is wrong and record name is just "root" for some reason?). There's also no builder method available for us to manually set an Avro schema, either, which I think we'll have to add to Beam.